extensions/cron/README.md

```diff
@@ -0,0 +1,112 @@
+# @loopback/cron
+
+This module contains a component to provide integration with
+[Cron](https://github.com/kelektiv/node-cron) so that applications can schedule
+jobs using `cron` based schedule.
+
+## Installation
+
+```sh
+npm install --save @loopback/cron
+```
+
+## Basic use
+
+### Register the Cron component
+
+The component should be loaded in the constructor of your custom Application
+class.
+
+Start by importing the component class:
+
+```ts
+import {CronComponent} from '@loopback/cron';
+```
+
+In the constructor, add the component to your application:
+
+```ts
+this.component(CronComponent);
+```
+
+### Register Cron jobs
+
+We can create an instance of `CronJob` and bind it to the application context so
+it can be managed by the `CronComponent` life cycle.
+
+```ts
+import {CronJob, asCronJob} from '@loopback/cron';
+
+const job = new CronJob({
+  cronTime: '*/1 * * * * *', // Every one second
+  onTick: () => {
+    // Do the work
+  },
+  runOnInit: true, // Start the job immediately
+});
+app
+  .bind('cron.jobs.job1')
+  .to(job)
+  .apply(asCronJob);
+```
+
+Alternatively, we can also define a provider class:
+
+```ts
+import {CronJob, cronJob, CronJobConfig} from '@loopback/cron';
+import {config, Provider, createBindingFromClass} from '@loopback/context';
+
+@cronJob()
+class CronJobProvider implements Provider<CronJob> {
+  constructor(@config() private jobConfig: CronJobConfig = {}) {}
+  value() {
+    const job = new CronJob({
+      // ... default config
+      ...this.jobConfig,
+    });
+    return job;
+  }
+}
+
+const jobBinding = createBindingFromClass(CronJobProvider);
+app.add(jobBinding);
+const now = new Date();
+now.setMilliseconds(now.getMilliseconds() + 10);
+const jobConfig: CronJobConfig = {
+  name: 'job-B', // Name the job
+  cronTime: sendAt(now), // Start the job in 10 ms
```

Comment:
I love the concept of running cron jobs from a loopback application.
Can we have another example besides one that executes every 10 seconds. Perhaps, register a job to occur at midnight everyday with a name like 'produce report' or something along those lines? Just a thought.

Reply:
The job is for testing purpose and we can not wait for midnight for a CI run. :-).